src/main/java/ghidrathon/interpreter/GhidrathonInterpreter.java

```diff
@@ -193,10 +194,19 @@ private void extendPythonSysPath() {
    */
   private void configureJepMainInterpreter() throws JepException, FileNotFoundException {
 
-    File ghidrathonSaveFile =
-        new File(
-            Application.getApplicationRootDirectory().getParentFile().getFile(false),
-            GhidrathonInterpreter.GHIDRATHON_SAVE_FILENAME);
+    // get the ghidrathon.save path from the environment variable or use the default path
+    String ghidrathonSavePath = System.getenv(GHIDRATHON_SAVE_PATH);
+    File ghidrathonSaveFile;
+
+    if (ghidrathonSavePath != null && !ghidrathonSavePath.isEmpty()) {
+      ghidrathonSaveFile = new File(ghidrathonSavePath, GHIDRATHON_SAVE_FILENAME);
+    } else {
+      ghidrathonSaveFile =
+          new File(
+              Application.getApplicationRootDirectory().getParentFile().getFile(false),
+              GHIDRATHON_SAVE_FILENAME);
+    }
+
```

Comment:
Let's add debug output that informs the user where the save path is coming from (environment variable or default). This will help users troubleshoot cases where stale environment variables are hanging around.

Reply:
This is now resolved in the latest commit for both Python and Java sides.